### PR TITLE
feat: display the date and time for a credential

### PR DESCRIPTION
**Pull request by 0xkelvin**
## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_identity/Cargo.toml [hunk 1]
```diff
@@ -79,6 +53,7 @@ rand = { version = "0.8", default-features = false }
 tracing = { version = "0.1", default_features = false }
 hex = { version = "0.4", default-features = false }
 async-trait = "0.1.64"
+time = { version = "0.3.20", features = ["macros", "formatting"]}
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_identity/Cargo.toml]: Could you please keep that line without the reformatting above so we keep the diff minimal?

File: implementations/rust/ockam/ockam_identity/src/credential.rs [hunk 2]
```diff
@@ -64,9 +68,35 @@ impl fmt::Display for Credential {
             " Issuer: {} ({})",
             credential_data.issuer, credential_data.issuer_key_label
         )?;
-        //TODO: write timestamps on human-readable format. Should we add a dependency for this?
-        writeln!(f, " Created: {}", u64::from(credential_data.created))?;
-        writeln!(f, " Expires: {}", u64::from(credential_data.expires))?;
+        
+        let human_readable_time = |time: Timestamp| {
```
Comment (etorreborre) [inline implementations/rust/ockam/ockam_identity/src/credential.rs]: That's great thanks! What would be even better would be a unit test for this `Display` instance if you have a bit of time :-). This way we can clearly see what's the expected formatting (and avoid regressions)

File: implementations/rust/ockam/ockam_identity/Cargo.toml (hunk 1 above)
Comment (0xkelvin) [reply implementations/rust/ockam/ockam_identity/Cargo.toml]: this line i added new for https://github.com/time-rs/time dependency

File: implementations/rust/ockam/ockam_identity/src/credential.rs (hunk 2 above)
Comment (0xkelvin) [reply implementations/rust/ockam/ockam_identity/src/credential.rs]: let me try to refactor according to what you have mentioned in bonus points, then i will write unit test

File: implementations/rust/ockam/ockam_identity/Cargo.toml (hunk 1 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_identity/Cargo.toml]: Indeed, I was just mentioning the reformatting of [the lines above](https://github.com/build-trust/ockam/pull/4429/files#diff-3b48031176edab1fd6c3d7444fd835b76b99afa476a67a6f0e17835b27b269dfL21).

File: implementations/rust/ockam/ockam_identity/src/credential.rs (hunk 2 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_identity/src/credential.rs]: Thank you

**Comment by etorreborre** [comment]
Hi @0xkelvin I rebased your PR and added a unit test for the display behaviour. I chose to do the following:
 
 - remove the `---` delimiters and leave that responsibility to the callers of the `Credential` or `CredentialData` users
 - align the fields values

@CAOakleyII I know that there are plans for making the display of data more uniform across the command line. Please tell us if you wish to deal with those 2 display instances differently.

**Comment by caoakleyii** [comment]
> Hi @0xkelvin I rebased your PR and added a unit test for the display behaviour. I chose to do the following:
> 
> * remove the `---` delimiters and leave that responsibility to the callers of the `Credential` or `CredentialData` users
> * align the fields values
> 
> @CAOakleyII I know that there are plans for making the display of data more uniform across the command line. Please tell us if you wish to deal with those 2 display instances differently.

Thanks for keeping me in the loop! This works well for now, I don't believe there's a needed reason for the `---` anywhere at this point.

**Review by 0xkelvin — COMMENTED** [review]
LGTM

**Review by etorreborre — DISMISSED** [review]

**Review by metaclips — APPROVED** [review]

**Pull request merged**